src/worker.js

```diff
@@ -124,6 +124,25 @@ self.onmessage = (e) => {
       var imports = {};
 #endif
 
+    // Until we initialize the runtime, queue up any further incoming messages.
+    let messageQueue = [];
+    self.onmessage = (e) => messageQueue.push(e);
+
+    // And add a callback for when the runtime is initialized.
+    self['startWorker'] = (instance) => {
```

Comment:
Why not `self.startWorker`?  (i.e. why use a string here but not for `self.onmessage` just above?)

Reply:
Because this is a custom property instead of a standard one; like with other properties in the codebase, Closure doesn't like it if written without `obj['prop']` syntax.

Reply:
I see so its just for closure's benefit?   Would it be better if closure could minify this (internal) name?   Why use `self` at all here?  Why not just use a bare global var?  Or globalThis?

Reply:
> I see so its just for closure's benefit?

Yeah.

> Would it be better if closure could minify this (internal) name?

Ideally, yes, but it complains about unknown property instead of trying to minify it + even if it could, it can't reason about cross-file usages in our case.

> Why use self at all here? Why not just use a bare global var?

Well you can't use a string as a var, and if I declare a var in worker, then once again Closure won't know it's defined in e.g. preamble, so `self` is just an easier escape hatch.

`globalThis` would also work, sure, I just used `self` as a matter of habit because `self` was there ~forever and we already use it in other places in the codebase, while `globalThis` is a relatively recent JS feature and might be not supported in some older browsers Emscripten maybe still cares about?

FWIW in https://github.com/emscripten-core/emscripten/commit/3935cdcc57409e05830be9603e8eb0b906809f40 I tried a slightly different approach with defining the variable in closure-externs instead. I can switch to that here if that's what you prefer.

Reply:
Extracted the other commit into https://github.com/emscripten-core/emscripten/pull/18313, so it will kill both issues with one stone.